src/pool/poolClient.ts

```diff
@@ -177,57 +179,49 @@ export class PoolEventState {
     };
   }
   getL1TokenFromReceipt(receipt: TransactionReceipt): string {
-    const events = receipt.logs
-      .map((log) => {
-        try {
-          return this.makeEventFromLog(log);
-        } catch (err) {
-          // return nothing, this throws a lot because logs from other contracts are included in receipt
-          return undefined;
-        }
-      })
-      // filter out undefined
-      .filter(exists);
-
-    const eventState = hubPool.getEventState(events);
+    const events = receipt.logs.map((log) => {
+      try {
+        return this.makeEventFromLog(log);
+      } catch (err) {
+        // return nothing, this throws a lot because logs from other contracts are included in receipt
+        return undefined;
+      }
```

Comment:
Instead of ignoring errors, could we use the address associated with each log to filter events that were from our contract and only try to parse those? If we only parse those, this would only error if our abi was wrong (which is something we'd want to know!).

Something like:

```ts
const events = receipt.logs
    .filter((log) => 
        ethers.utils.getAddress(log.address) === ethers.utils.getAddress(this.contract.address)
    )
    .map((log) => {
...
```

Reply:
good idea

Reply:
i've added this in both spots